pkg/controllers/v1alpha1/goosefs/goosefs_runtime_controller.go

```diff
@@ -0,0 +1,107 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package goosefs
+
+import (
+	"context"
+
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
+	"github.com/fluid-cloudnative/fluid/pkg/common"
+	"github.com/fluid-cloudnative/fluid/pkg/controllers"
+	"github.com/fluid-cloudnative/fluid/pkg/ddc/base"
+	cruntime "github.com/fluid-cloudnative/fluid/pkg/runtime"
+	"github.com/fluid-cloudnative/fluid/pkg/utils"
+)
+
+// Use compiler to check if the struct implements all the interface
+var _ controllers.RuntimeReconcilerInterface = (*RuntimeReconciler)(nil)
+
+// RuntimeReconciler reconciles a GooseFSRuntime object
+type RuntimeReconciler struct {
+	Scheme  *runtime.Scheme
+	engines map[string]base.Engine
+	mutex   *sync.Mutex
+	*controllers.RuntimeReconciler
+}
+
+// NewRuntimeReconciler create controller for watching runtime custom resources created
+func NewRuntimeReconciler(client client.Client,
+	log logr.Logger,
+	scheme *runtime.Scheme,
+	recorder record.EventRecorder) *RuntimeReconciler {
+	r := &RuntimeReconciler{
+		Scheme:  scheme,
+		mutex:   &sync.Mutex{},
+		engines: map[string]base.Engine{},
+	}
+	r.RuntimeReconciler = controllers.NewRuntimeReconciler(r, client, log, recorder)
+	return r
+}
+
+// Reconcile reconciles goosefs runtime
+// +kubebuilder:rbac:groups=data.fluid.io,resources=goosefsruntimes,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=data.fluid.io,resources=goosefsruntimes/status,verbs=get;update;patch
+
+func (r *RuntimeReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := cruntime.ReconcileRequestContext{
+		Context:        context.Background(),
+		Log:            r.Log.WithValues("goosefsruntime", req.NamespacedName),
+		NamespacedName: req.NamespacedName,
+		Recorder:       r.Recorder,
+		Category:       common.AccelerateCategory,
+		RuntimeType:    runtimeType,
+		Client:         r.Client,
+		FinalizerName:  runtimeResourceFinalizerName,
+	}
+
+	ctx.Log.V(1).Info("process the request", "request", req)
+
+	//	1.Load the Runtime
+	runtime, err := r.getRuntime(ctx)
```

Comment:
`getRuntime` there is no need for encapsulation

Reply:
Sorry, i am not clear your question. Can you describe more details? Thanks a lot.

Reply:
What I mean is that `getRuntime ` is a simple logic, and we don't really need to wrap a function

Reply:
Though the logic is sample, but i think warp function is ok for me : ) .  WDYT @cheyang

Reply:
i think getRuntime is need for encapsulation because every developer need to implement their own getRuntime function